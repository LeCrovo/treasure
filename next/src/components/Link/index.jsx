/* eslint-disable react/jsx-props-no-spreading */
import { memo } from 'react'
import NextLink from 'next/link'
import styled from 'styled-components'

const ColouredLink = styled(NextLink)`
  color: #6cc0e5;

  &:hover {
    color: #e6df55;
  }

  &:active {
    color: #999540;
  }

  &:visited {
    color: #e68393;
  }

  &:visited:hover {
    color: #e6df55;
  }
`

function Link({ children, ...props }) {
  return (
    <ColouredLink {...props}>{children}</ColouredLink>
  )
}

export default memo(Link)
