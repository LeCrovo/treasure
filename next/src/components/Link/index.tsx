import NextLink from 'next/link'
import { memo, type ReactNode } from 'react'
import styled from 'styled-components'

const ColouredLink = styled(NextLink)`
  color: #6cc0e5;

  &:hover {
    color: #e6df55;
  }

  &:active {
    color: #999540;
  }

  &:visited {
    color: #e68393;
  }

  &:visited:hover {
    color: #e6df55;
  }
`

interface InputGroupProps extends React.ComponentPropsWithoutRef<'a'> {
  children: ReactNode;
  href: string;
}

function Link(
  { children, ...props }:
  InputGroupProps,
) {
  return (
    <ColouredLink {...props}>{children}</ColouredLink>
  )
}

export default memo(Link)
