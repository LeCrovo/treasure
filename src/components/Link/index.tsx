import NextLink from 'next/link'
import { memo, type ReactNode } from 'react'

import styles from './styles.module.css'

interface InputGroupProps extends React.ComponentPropsWithoutRef<'a'> {
  children: ReactNode;
  href: string;
}

function Link(
  { children, ...props }:
  InputGroupProps,
) {
  return (
    <NextLink className={styles.link} {...props}>{children}</NextLink>
  )
}

export default memo(Link)
