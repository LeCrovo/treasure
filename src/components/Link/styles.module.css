.link {
  color: #6cc0e5;
}

.link:hover {
  color: #e6df55;
}

.link:active {
  color: #999540;
}

.link:visited {
  color: #e68393;
}

.link:visited:hover {
  color: #e6df55;
}
