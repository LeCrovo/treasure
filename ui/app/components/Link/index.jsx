import React, { memo } from 'react';
import { Link as DomLink } from 'react-router-dom';
import styled from 'styled-components';

const ColouredLink = styled(DomLink)`
  color: #6cc0e5;

  &:hover {
    color: #e6df55;
  }

  &:active {
    color: #999540;
  }

  &:visited {
    color: #e68393;
  }

  &:visited:hover {
    color: #e6df55;
  }
`;

function Link(props) {
  return <ColouredLink {...props} />;
}

export default memo(Link);
