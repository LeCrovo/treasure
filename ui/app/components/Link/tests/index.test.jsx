/* eslint-disable jsx-a11y/anchor-is-valid */
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';

import Link from '../index';

describe('<Link />', () => {
  test('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <BrowserRouter>
        <Link to="some-internal-link" />
      </BrowserRouter>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  test('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <BrowserRouter>
        <Link />
      </BrowserRouter>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
