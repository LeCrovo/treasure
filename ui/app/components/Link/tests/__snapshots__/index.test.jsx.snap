// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> Should render and match the snapshot 1`] = `
<a
  class="Link__ColouredLink-pyilfz-0 hkoxpZ"
  href="/some-internal-link"
/>
`;
